src/odemis/dataio/tiff.py

```diff
@@ -989,9 +989,19 @@ def _updateMDFromOME(root, das):
         # the ome stores the Z position in the metadata of each plane
         # but odemis MD_POS is the centre of the image
         if z_positions:
-            for da in das:
+            z_center = statistics.median(z_positions)
+            for ifd in hd_2_ifd.flat:
```

Comment:
Maybe the above if block from 976 can be combined with 993?

Reply:
It's not possible because that block is called immediately after parsing each image "plane". However, it's possible to compute the center only once we have parsed all the z-positions. So this has to be done *after* the for loop.